jupyter_collaboration/handlers.py

```diff
@@ -129,10 +86,86 @@ class YDocWebSocketHandler(WebSocketHandler, JupyterHandler):
        receiving a message.
     """
 
-    saving_document: Optional["asyncio.Task[Any]"]
+    files: Dict[str, FileLoader] = {}
     websocket_server: Optional[JupyterWebsocketServer] = None
     _message_queue: "asyncio.Queue[Any]"
 
+    def __init__(self, app: ServerWebApplication, request: HTTPServerRequest, **kwargs):
+        super().__init__(app, request, **kwargs)
+
+        # CONFIG
+        self._file_id_manager = self.settings["file_id_manager"]
+        self._ystore_class = self.settings["collaborative_ystore_class"]
+        self._cleanup_delay = self.settings["collaborative_document_cleanup_delay"]
+        # self.settings["collaborative_file_poll_interval"]
+        # self.settings["collaborative_document_save_delay"]
+
+        # Instantiate the JupyterWebsocketServer as a Class property
+        # if it doesn't exist yet
+        if self.websocket_server is None:
+            for k, v in self.config.get(self._ystore_class.__name__, {}).items():
+                setattr(self._ystore_class, k, v)
+
+            YDocWebSocketHandler.websocket_server = JupyterWebsocketServer(
+                rooms_ready=False,
+                auto_clean_rooms=False,
+                ystore_class=self._ystore_class,
+                log=self.log,
+            )
+
+        self._message_queue = asyncio.Queue()
+
+        # Get room
+        self._room_id: str = request.path.split("/")[-1]
```

Comment:
Nothing in particular. It was just to separate the initialization of the client.

> with potential ambiguities if there are / in the room id?

There should not be any ambiguities. We are using IDs for rooms instead of file paths.